### PR TITLE
NEW: Add recipe for astra-toolbox

### PR DESCRIPTION
**Pull request by carterbox**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Resolves #13886

### CODE REVIEW

**Comment by carterbox** [comment]
@conda-forge/help-python-c Ready for review. 😁

**Comment by carterbox** [comment]
@conda-forge/help-python-c Ready for review.

**Comment by carterbox** [comment]
@conda-forge/help-python, Ready for review.

**Comment by jakirkham** [comment]
cc @conda-forge/staged-recipes

File: recipes/astra-toolbox/bld.bat [hunk 1]
```diff
@@ -0,0 +1,33 @@
+@echo off
+
+set R=%SRC_DIR%
+
+set B_VC=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\
+rem call "%B_VC%\vcvars64.bat" amd64 8.1 -vcvars_ver=14.0
+
+set B_BV=1_69
+set B_BOOST=E:\wjp\boost_%B_BV%_0
```
Comment (isuruf) [inline recipes/astra-toolbox/bld.bat:9]: Why?

File: recipes/astra-toolbox/bld.bat [hunk 2]
```diff
@@ -0,0 +1,33 @@
+@echo off
+
+set R=%SRC_DIR%
+
+set B_VC=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\
+rem call "%B_VC%\vcvars64.bat" amd64 8.1 -vcvars_ver=14.0
```
Comment (isuruf) [inline recipes/astra-toolbox/bld.bat:6]:
```suggestion
```

File: recipes/astra-toolbox/build.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+case `uname` in
+  Darwin*)
+    CC="gcc -stdlib=libstdc++"
```
Comment (isuruf) [inline recipes/astra-toolbox/build.sh:5]: Please use `clang` and `libc++` on macos

File: recipes/astra-toolbox/bld.bat (hunk 1 above)
Comment (carterbox) [reply recipes/astra-toolbox/bld.bat:9]: I don't know, I was trying to accomplish the build with the least amount of work by copying the provided build scripts from the source repo. 😅

File: recipes/astra-toolbox/build.sh (hunk 3 above)
Comment (carterbox) [reply recipes/astra-toolbox/build.sh]: Deleted these lines. Not building for OSX anyways.

File: recipes/astra-toolbox/bld.bat (hunk 2 above)
Comment (carterbox) [reply recipes/astra-toolbox/bld.bat]: Deleted these lines, they don't seem necessary in the conda-forge environment. They are from the source conda recipe.

**Comment by carterbox** [comment]
@conda-forge/staged-recipes, ready for review

**Comment by carterbox** [comment]
@conda-forge/help-python-c, ready for review

**Comment by carterbox** [comment]
@conda-forge/help-python-c Ready for review.

File: recipes/astra-toolbox/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  skip: true  # [not (linux64 or win64)]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - cython >=0.13
+    # libastra includes boost in its public headers
+    - boost-cpp
+    - libastra =={{ version.split('.')[:2]|join('.') }}
```
Comment (beckermr) [inline recipes/astra-toolbox/meta.yaml:23]: Is there a reason libastra doesn't have a run export?

**Review by beckermr — CHANGES_REQUESTED** [review]

File: recipes/astra-toolbox/meta.yaml (hunk 4 above)
Comment (carterbox) [reply recipes/astra-toolbox/meta.yaml:23]: The libastra recipe does have a run_export, but run_exports don't pin in the host section, they pin in the run section.

**Comment by carterbox** [comment]
@conda-forge/help-python-c, please review.

**Review by beckermr — CHANGES_REQUESTED** [review]

File: recipes/astra-toolbox/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    - numpy
+    - python
+    - setuptools
+    - scipy
+    - six
+  run:
+    - {{ pin_compatible('libastra', min_pin='x.x', max_pin='x.x') }}
+    - {{ pin_compatible('numpy', min_pin='x.x', max_pin='x') }}
```
Comment (carterbox) [inline recipes/astra-toolbox/meta.yaml:31]:
```suggestion
    - {{ pin_compatible('numpy') }}
```

**Comment by carterbox** [comment]
@conda-forge/help-python-c, please review.

**Pull request merged**